### PR TITLE
Backport ccompat refactor

### PR DESCRIPTION
**Pull request by carlesarnal**
Backport of https://github.com/Apicurio/apicurio-registry/pull/3619 to 2.5.x

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectVersionsResourceImpl.java [hunk 1]
```diff
@@ -46,42 +60,90 @@
...
             String version) throws Exception {
-
-        return facade.getSchema(subject, version);
+        return getSchema(null, subject, version, false);
     }
 
     @Override
     @Audited(extractParameters = {"0", KEY_ARTIFACT_ID, "1", KEY_VERSION})
     @Authorized(style=AuthorizedStyle.ArtifactOnly, level=AuthorizedLevel.Write)
     public int deleteSchemaVersion(
```
Comment (jsenko) [inline app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectVersionsResourceImpl.java:126]: Also similar code

File: app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectsResourceImpl.java [hunk 2]
```diff
@@ -41,22 +54,41 @@
...
+    @Authorized(style = AuthorizedStyle.None, level = AuthorizedLevel.Read)
     public List<String> listSubjects() {
-        return facade.getSubjects(false);
+        return getStorage().searchArtifacts(Set.of(SearchFilter.ofGroup(null)), OrderBy.createdOn, OrderDirection.asc, 0, getCconfig().getMaxSubjects().get()).getArtifacts().stream().filter(searchedArtifactDto -> isCcompatManagedType(searchedArtifactDto.getType()) && shouldFilterState(false, searchedArtifactDto.getState())).map(SearchedArtifactDto::getId).collect(Collectors.toList());
     }
 
     @Override
-    @Authorized(style=AuthorizedStyle.ArtifactOnly, level=AuthorizedLevel.Read)
+    @Authorized(style = AuthorizedStyle.ArtifactOnly, level = AuthorizedLevel.Read)
     public Schema findSchemaByContent(String subject, SchemaInfo request) throws Exception {
```
Comment (jsenko) [inline app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectsResourceImpl.java:64]: Also similar code

File: app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java
Comment (jsenko) [inline app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java:1]: Why remove this?

Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java:1]: I refactored it in main and we only have v7, and the logic is moved into the resources. It only added complexity at that point.

File: app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectVersionsResourceImpl.java [hunk 3]
```diff
@@ -46,42 +60,90 @@
...
     public List<Integer> listVersions(String subject) throws Exception {
-        return facade.getVersions(subject);
-    }
+            return getStorage().getArtifactVersions(null, subject).stream().map(VersionUtil::toLong).map(getConverter()::convertUnsigned).sorted().collect(Collectors.toList());
+        }
 
     @Override
     @Audited(extractParameters = {"0", KEY_ARTIFACT_ID})
     @Authorized(style=AuthorizedStyle.ArtifactOnly, level=AuthorizedLevel.Write)
     public SchemaId register(String subject, SchemaInfo request) throws Exception {
```
[the remark this answers is not in the document]
Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/ccompat/rest/v6/impl/SubjectVersionsResourceImpl.java:73]: Yes, you're going to find quite some duplicates here. I just did the "minimum" so I could backport the functionality, so a bunch of duplicates are guaranteed for sure. I can address them, but this code is gone on main, so there are no duplicates either.

**Comment by jsenko** [comment]
@carlesarnal Makes sense if the code on `main` is different. I guess it depends on how much maintenance will we do in the `2.5.x` branch in the future. If you think the refactoring is not worth the effort, I'm ok with that.

**Comment by carlesarnal** [comment]
> @carlesarnal Makes sense if the code on `main` is different. I guess it depends on how much maintenance will we do in the `2.5.x` branch in the future. If you think the refactoring is not worth the effort, I'm ok with that.

I would say we can go ahead with this and if we have to extend the support for 2.5.x we can refactor this, it's a pretty trivial one, just tedious.

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**